File: frontend/src/types/health.ts
import type { ErrorHistory } from './error'

export interface HealthResponse {
  status: 'ok'
  service: string
  version: string
}

export interface AgentInfo {
  name: string
  hostname: string
  local_hostname: string
  ip: string
  port: number
  status: 'running'
}

export type ConnectionState = 'connected' | 'disconnected' | 'reconnecting' | 'discovering'

export interface AgentConnectedEvent {
  type: 'agent.connected'
  timestamp: string
  data: {
    service: string
    version: string
  }
}

export interface PongEvent {
  type: 'pong'
  timestamp: string
}

export interface ErrorDetectedEvent {
  type: 'error.detected' | 'error.updated'
  timestamp: string
  data: ErrorHistory
}

export type AgentEvent = AgentConnectedEvent | PongEvent | ErrorDetectedEvent

File: backend/app/schemas/health.py
from typing import Literal

from pydantic import BaseModel


class HealthResponse(BaseModel):
    status: Literal["ok"]
    service: str
    version: str


class AgentInfoResponse(BaseModel):
    name: str
    hostname: str
    local_hostname: str
    ip: str
    port: int
    status: Literal["running"]
